src/core/sagas/utils.js

```diff
@@ -1,19 +1,24 @@
 /* @flow */
 import { ErrorHandler } from 'core/errorHandler';
-import log from 'core/logger';
+import defaultLog from 'core/logger';
 import type { ApiStateType } from 'core/reducers/api';
 
-export function createErrorHandler(id: string): typeof ErrorHandler {
+
+export function createErrorHandler(
+  id: string,
+  { log = defaultLog }: {| log: typeof defaultLog |} = {}
+): typeof ErrorHandler {
   return new ErrorHandler({
     id,
     // Make sure the dispatch() method can't be used. A saga will yield
     // put(action) instead so this shouldn't cause a problem.
-    dispatch: () => log.error(
-      'ErrorHandler cannot dispatch from a saga'),
+    dispatch: () => log.error('ErrorHandler cannot dispatch from a saga'),
   });
 }
 
-// Convenience function to extract API info.
-export function getApi(state: {| api: ApiStateType |}): ApiStateType {
-  return state.api;
+// Convenience function to extract state info.
+export function getState(
+  state: {| api: ApiStateType, auth: Object |}
+): Object {
```

Comment:
I find flow's error outputs to be hard to follow as well, but I see what that's doing.